File: 217-ContainsDuplicate/217-ContainsDuplicate.cpp
// Last updated: 7/7/2026, 2:57:25 pm
1class Solution {
2public:
3    bool containsDuplicate(vector<int>& nums) {
4        unordered_set<int>st;
5        for(int i=0;i<nums.size();i++){
6            if(st.find(nums[i])!=st.end()){
7                return true;
8            }
9            st.insert(nums[i]);
10        }        
11        return false;
12    }
13};

File: 217-ContainsDuplicate/217-ContainsDuplicate.py
# Last updated: 29/8/2026, 7:39:30 pm
1class Solution:
2    def containsDuplicate(self, nums: List[int]) -> bool:
3        visited = dict()
4        for num in nums:
5            if num in visited:
6                return True
7            visited[num]=True
8        return False
9
10        # seen = set()
11        # for num in nums:
12        #     if num in seen:
13        #         return True
14        #     seen.add(num)
15        # return False
16
17